frontend/src/pages/Wallet.js

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: MIT
+// SDPX-FileCopyrightText: 2021 Que Le <b.le@tu-berlin.de>
+
+import {
+	Button,
+	Table,
+	TableBody,
+	TableCell,
+	TableContainer,
+	TableRow,
+} from "@mui/material";
+import React, { useEffect } from "react";
+import appwriteApi from "../api/appwriteApi";
+import { useHistory } from "react-router-dom";
+import Grid from "@mui/material/Grid";
+import { useState } from "react";
+import detectEthereumProvider from "@metamask/detect-provider";
+import CenterFlexBoxMedium from "../components/CenterFlexBoxMedium";
+const { utils } = require( "ethers" );
+// import utils from "ethers";
+
+function AccountEntry({ data }) {
+	return 	<TableRow>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{data.pubAddress}
+		</TableCell>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{utils.formatEther(data.balance)}
+		</TableCell>
+	</TableRow>;
+}
+
+function EthAccountsDetail({ publicAddresses }) {
+	const [detailData, setDetailData] = useState([]);
+
+	async function fetchAccountsDetails(publicAddresses) {
+		const provider = await detectEthereumProvider();
+		if (!provider) {
+			console.error("no ethereum provider");
+			return;
+		}
+		let accDetail = [];
```

Comment:
It would be nice if we can use variables that can be communicated well, and even understood by those who might not be even programmers. Abbreviations also have the problem that they can have multiple meanings like `accent`, `accident`, `acceleration`, `acceptance`, `accuracy`. I think, `acc` is not even a common abbreviation for `account`.

Another issue: a variable which contains a collection of things is singular here. I think a plural name would fit more like `accountDetails`. Even a better fit: `accountDetailList`.

Reply:
I disagree with the acc part. First, code does not need to be understood by non programmers. Second, the variable is defined in the context of a function called `fetchAccountsDetails`, i.e. it is very obvious that acc does not mean accent, accident or anything else, but means account. I agree that it should be plural.

Reply:
Yeah, that's right. From the context, it's clear. I just don't like such abbreviations because it doesn't make sense to abbreviate short names just for the sake of it. It provides no advantage and makes it more difficult to talk about variable. On the other hand, if actual english words are used, it has the advantage that even unfamiliar or non-programming people could know what is talked about. This is at least useful for companies where manager roles work more closely together with developer roles.